### PR TITLE
Features/m2 1315 more filters

### PR DESCRIPTION
**Pull request by kdanilenka32**

### CODE REVIEW

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx [hunk 1]
```diff
@@ -2,41 +2,90 @@ import { Box, Button } from '@mui/material';
...
+    defaultValues: {
+      startDateEndDate: [],
+      moreFiltersVisisble: false,
+      startTime: '',
+      endTime: '',
+    },
+  });
+
+  const moreFiltersVisisble = watch('moreFiltersVisisble');
+  const filterByIdentifier = watch('filterByIdentifier');
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx:24]: don't we have a default value for this field?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx [hunk 2]
```diff
@@ -2,41 +2,90 @@ import { Box, Button } from '@mui/material';
...
-        control={control}
-        label={t('endTime')}
-        wrapperSx={{ width: '13rem', marginRight: theme.spacing(1.2) }}
-      />
-      <Button>{t('moreFilters')}</Button>
-    </StyledForm>
+        <Button
+          onClick={() => setValue('moreFiltersVisisble', !moreFiltersVisisble)}
+          sx={{
+            height: '5.5rem',
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx:54]: don't we follow the rule to add sx styles only for margins and paddings

File: src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx [hunk 3]
```diff
@@ -69,22 +69,24 @@ export const TagsInputController = <T extends FieldValues>({
...
-                        handleToggleSelectAll();
-                      }}
-                      label={labelAllSelect}
-                      sx={{
-                        width: '100%',
-                      }}
-                      control={<Checkbox id="select-all-checkbox" checked={selectedAll} />}
-                    />
-                  </ListItem>
+                  {options?.length ? (
```
Comment (alex-i-e) [inline src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx:72]: no need in ternary operator :
```suggestion
                  {Boolean(options?.length) ?? (
```

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx [hunk 4]
```diff
@@ -2,41 +2,90 @@ import { Box, Button } from '@mui/material';
...
-    control,
-    inputSx: { width: '19rem' },
-  };
+  const { t } = useTranslation('app');
+  const { control, setValue, watch } = useForm<FilterValues>({
+    defaultValues: {
+      startDateEndDate: [],
+      moreFiltersVisisble: false,
+      startTime: '',
+      endTime: '',
```
Comment (alex-i-e) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx:19]: don't we have a default value (with empty array) for `identifier` and `versions` ?

Comment (kdanilenka32) [reply src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx:19]: no

File: src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx [hunk 5]
```diff
@@ -69,22 +69,24 @@ export const TagsInputController = <T extends FieldValues>({
...
+                    <ListItem
+                      onMouseDown={(e) => e.preventDefault()} // prevent blur
+                      sx={{ pl: theme.spacing(2.8) }}
+                    >
+                      <FormControlLabel
+                        onClick={(e) => {
+                          e.preventDefault(); // prevent blur
```
[the remark this answers is not in the document]
Comment (kdanilenka32) [reply src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx]: fixed

File: src/shared/components/DatePicker/DatePicker.tsx [hunk 6]
```diff
@@ -65,35 +66,60 @@ export const DatePicker = <T extends FieldValues>({
...
           }
 
           return getStringFromDate(singleDate) || '';
         };
 
+        const textFieldProps = {
+          fullWidth: true,
+          disabled: true,
+          onClick: handlePickerShow,
+          className: (open && 'active') || '',
```
[the remark this answers is not in the document]
Comment (kdanilenka32) [reply src/shared/components/DatePicker/DatePicker.tsx]: fixed

File: src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx (hunk 3 above)
Comment (kdanilenka32) [reply src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx:81]: there is not much difference

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx (hunk 4 above)
Comment (alex-i-e) [reply src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx:19]: why not, let's determine default values

File: src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx (hunk 3 above)
Comment (alex-i-e) [reply src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx:81]: `: null` is redundant

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx (hunk 2 above)
Comment (alex-i-e) [reply src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx:54]: @IhorNesterenko99, could you add any details?

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx (hunk 1 above)
Comment (alex-i-e) [reply src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx:24]: why not?

File: src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx (hunk 3 above)
Comment (kdanilenka32) [reply src/shared/components/FormComponents/TagsAutocompleteController/TagsAutocompleteController.tsx:81]: type conversion is redundant

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx (hunk 4 above)
Comment (kdanilenka32) [reply src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ReportFilters/ReportFilters.tsx:19]: it is conditional values

**Review by alex-i-e — APPROVED** [review]

File: src/shared/components/FormComponents/Switch/Switch.styles.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import { Switch, styled } from '@mui/material';
+
+import { variables } from 'shared/styles';
+
+export const StyledSwitch = styled(Switch)`
```
Comment (nmialik) [inline src/shared/components/FormComponents/Switch/Switch.styles.ts:5]: maybe it's better to override styles for mui switch in theme file?

**Pull request merged**